### ISSUE TITLE
`go mod` failures should not block PRs

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#25081]
`go mod` blocks an unrelated PR if a Go module we have a dependency on was updated recently. This causes non-Go PRs to be blocked. [Example here.](https://buildkite.com/sourcegraph/sourcegraph/builds/108655#112a07e5-34bc-4413-8d68-7d7b7d3c5b01) 

This should not happen as it is not a problem with the PR itself. It requires PR authors to add changes to the PR that are beyond the scope of the PR, or wait for another PR that updates the Go modules to be completed first to unblock the original PR.

As a first step, `go mod` linter failure could only block PRs if the PR has Go files in it. As an even better alternative, `go mod` should be updated similar to how our `yarn` dependencies are kept up-to-date, using a bot like `renovate` to generate a PR with the update.

**Comment by camdencheek** [comment]
I have a feeling this is because, in our `generate` directives, we build and run `go build` and `go install` on `golang.org/x/tools`. I think the solution here is to pin the version so these don't change when `latest` changes, which also forces `go install` to use "module-aware mode", keeping it from mutating `go.mod`.

**Comment by limitedmage** [comment]
@ggilmore Does #25082 fix this? Or is there more work to be done to resolve this issue completely?

**Comment by bobheadxi** [comment]
I haven't seen this in a while, so presumably https://github.com/sourcegraph/sourcegraph/pull/25082 did the trick - closing
